packages/shared/lib/clients/oauth2.client.ts

```diff
@@ -18,10 +18,9 @@ export function getSimpleOAuth2ClientConfig(
     connectionConfig: Record<string, string>
 ): Merge<ModuleOptions, { http: WreckHttpOptions }> {
     const templateTokenUrl = typeof template.token_url === 'string' ? template.token_url : (template.token_url!['OAUTH2'] as string);
-    const strippedTokenUrl = templateTokenUrl.replace(/connectionConfig\./g, '');
-    const tokenUrl = new URL(interpolateString(strippedTokenUrl, connectionConfig));
-    const strippedAuthorizeUrl = template.authorization_url!.replace(/connectionConfig\./g, '');
-    const authorizeUrl = new URL(interpolateString(strippedAuthorizeUrl, connectionConfig));
+    const tokenUrl = new URL(interpolateString(templateTokenUrl.replace(/connectionConfig\./g, ''), encodeParameters(connectionConfig)));
+    const authorizeUrl = new URL(interpolateString(template.authorization_url!.replace(/connectionConfig\./g, ''), encodeParameters(connectionConfig)));
```

Comment:
logic looks ok but it is quite hard to read. what about having a local function like this:
```
function makeUrl(template: string, config: Record<string, any>): URL {
  const cleanTemplate = template.replace(/connectionConfig\./g, '');
  const encodedParams = encodeParameters(config);
  const interpolatedUrl = interpolateString(cleanTemplate, encodedParams);
  return new URL(interpolatedUrl);
}

const tokenUrl = makeUrl(templateTokenUrl, connectionConfig);
const authorizeUrl = makeUrl(template.authorization_url!, connectionConfig);
```

Reply:
Ah yes, making it more readable. I've updated the PR with the [latest](https://github.com/NangoHQ/nango/pull/2608/commits/bfe901e97884118a43eb4515b50788548c4f4bab) commit. Thank you🙏!